### PR TITLE
Contract tweaks

### LINKED ISSUE #1179: Handle Missing Contracts in Dispute Resolution
If a user starts a dispute after the other party has gone offline, and the other party does not come back online, the offline party's contract will not be supplied to the moderator. 

In that situation, clicking the resolve dispute button will return an error from the server unless 100% of the funds are awarded to the party that started the dispute.

Currently the error does not describe the reason it is happening in layman's terms, and there is no indication for the moderator that they can resolve the dispute by awarding the funds 100% to one party. 

Some options for dealing with that situation (suggestions welcome):

- provide an error that explains what needs to happen before the dispute can be resolved.
- if a contract is missing, let the moderator know they can resolve the dispute by awarding 100% of the funds to the party that started the dispute.
- disable the input for the party with the missing contract, or otherwise prevent the moderator from putting in percentages that will result in an error.

When the dispute resolution screen is shown, the order model can be checked for the buyer and vendor contracts if we want to indicate one is missing or otherwise change the UI.

### PR DESCRIPTION
**Pull request by rmisio**
closes #1179 
- adds an icon next to the contract menu item indicating whether a contract is unavailable and/or a contract has errors
- beefed up messaging on the contract tab + it will automatically update when the other contract becomes available

### CODE REVIEW

File: js/views/modals/orderDetail/contractTab/Contract.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    };
+
+    super(options);
+
+    if (!options.contract) {
+      throw new Error('Please provide a contract.');
+    }
+
+    this.contract = options.contract;
+    this.options = opts || {};
```
Comment (jjeffryes) [inline js/views/modals/orderDetail/contractTab/Contract.js]: Would opts ever be false?

File: js/views/modals/orderDetail/contractTab/ContractTab.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      ...options,
+    };
+
+    super(opts);
+
+    if (!this.model) {
+      throw new Error('Please provide a model.');
+    }
+
+    this.options = opts || {};
```
Comment (jjeffryes) [inline js/views/modals/orderDetail/contractTab/ContractTab.js]: Same question here, could opts ever be false?

File: js/models/order/ResolveDispute.js [hunk 3]
```diff
@@ -27,30 +43,50 @@ export default class extends BaseModel {
...
+        addError('vendorPercentage',
+          app.polyglot.t('resolveDisputeModelErrors.vendorPercentageMustBe100'));
+      } else {
+        vendorPercentageOk = true;
+      }
     } else {
-      vendorPercentageOk = true;
+      if (attrs.vendorPercentage !== 0) {
+        addError('vendorPercentage',
+          'The vendor amount must be zero if their contract is not available.');
```
[the remark this answers is not in the document]
Comment (rmisio) [reply js/models/order/ResolveDispute.js]: This phrase will never make it to our UI because we disable the field. The check is mainly there for data integrity and future-proofing.

File: js/models/order/ResolveDispute.js [hunk 4]
```diff
@@ -27,30 +43,50 @@ export default class extends BaseModel {
...
+        addError('buyerPercentage',
+          app.polyglot.t('resolveDisputeModelErrors.buyerPercentageMustBe100'));
+      } else {
+        buyerPercentageOk = true;
+      }
     } else {
-      buyerPercentageOk = true;
+      if (attrs.buyerPercentage !== 0) {
+        addError('buyerPercentage',
+          'The buyer amount must be zero if their contract is not available.');
```
[the remark this answers is not in the document]
Comment (rmisio) [reply js/models/order/ResolveDispute.js]: same as above.

**Comment by rmisio** [comment]
@jjeffryes is this one looking ok to merge? I wanted to use some of it's code in the Error Processing PR I'm working on.
